halva/src/main/java/io/soabase/halva/comprehension/MonadicFor.java

```diff
@@ -0,0 +1,43 @@
+package io.soabase.halva.comprehension;
+
+import io.soabase.halva.any.AnyVal;
+
+import java.util.function.Predicate;
+import java.util.function.Supplier;
+import java.util.function.Function;
+
+/** Interface class for the non-type-safe version "for comprehension" for Monadic classes

```

Comment:
> But only because you had 2 types T and R, I was going to ask why you didn't just have T?)

I think that might be a bug. I'll look at it. I had so many different implementations before I settled on this that the "R" could just be a vestige.